### PR TITLE
Share CBS Session

### PR DESCRIPTION
**Pull request by yunhaoling**
- [x] fix bug of sharing cbs session

### CODE REVIEW

File: uamqp/client.py [hunk 1]
```diff
@@ -871,7 +874,8 @@ def __init__(
         self._last_activity_timestamp = None
         self._was_message_received = False
         self._message_received_callback = None
-        self._received_messages = None
+        self._streaming_receive = False
+        self._received_messages = compat.queue.Queue()
```
Comment (annatisch) [inline uamqp/client.py]: These two lines don't seem to be in `client_async.close_async()`?

Comment (yunhaoling) [reply uamqp/client.py]: you mean resetting these two variables when a client is closed? but we don't support open-close-open..

File: uamqp/client.py [hunk 2]
```diff
@@ -270,6 +271,8 @@ def open(self, connection=None):
         if self._keep_alive_interval:
             self._keep_alive_thread = threading.Thread(target=self._keep_alive)
             self._keep_alive_thread.start()
+        if self._ext_connection:
+            connection.release()
```
Comment (annatisch) [inline uamqp/client.py]: I think this should go in a `try/finally` statement

File: uamqp/async_ops/client_async.py [hunk 3]
```diff
@@ -256,6 +256,8 @@ def __init__(
                 loop=self.loop)
         if self._keep_alive_interval:
             self._keep_alive_thread = asyncio.ensure_future(self._keep_alive_async(), loop=self.loop)
+        if self._ext_connection:
+            connection.release_async()
```
Comment (annatisch) [inline uamqp/async_ops/client_async.py]: Same comment re `try/finally`

**Pull request merged**